From 318cf4509f47ad80519d1905ac7673a3c516bb31 Mon Sep 17 00:00:00 2001
From: dholland <dholland@pkgsrc.org>
Date: Wed, 21 Oct 2026 23:38:31 +0000
Subject: [PATCH] archivers/libarchive: fix build on -10 / with older bmake

Avoid using $^ in the makefile. With bmake that doesn't support it,
which includes the one on -10 which is still supported, it expands to
empty and the result is that the build hangs trying to grep from
stdin.

Once -10 is EOL this change can be dropped.

Build fix, so no need to bump anything.

PR 60077
---
 archivers/libarchive/files/Makefile.am | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/archivers/libarchive/files/Makefile.am b/archivers/libarchive/files/Makefile.am
index 8bc70c5f752a..bf8d5be15759 100644
--- a/archivers/libarchive/files/Makefile.am
+++ b/archivers/libarchive/files/Makefile.am
@@ -786,7 +786,7 @@ libarchive_test_LDADD= $(LTLIBICONV)
 # above.
 libarchive/test/list.h: $(libarchive_test_SOURCES)
 	@$(MKDIR_P) libarchive/test
-	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $^ | LC_COLLATE=C sort > $@
+	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $(libarchive_test_SOURCES) | LC_COLLATE=C sort > $@
 
 libarchive_TESTS_ENVIRONMENT= LIBARCHIVE_TEST_FILES=`cd $(top_srcdir);/bin/pwd`/libarchive/test LRZIP=NOCONFIG
 
@@ -1469,7 +1469,7 @@ bsdtar_test_CPPFLAGS=\
 
 tar/test/list.h: $(bsdtar_test_SOURCES)
 	@$(MKDIR_P) tar/test
-	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $^ | LC_COLLATE=C sort > $@
+	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $(bsdtar_test_SOURCES) | LC_COLLATE=C sort > $@
 
 if BUILD_BSDTAR
 bsdtar_test_programs= bsdtar_test
@@ -1628,7 +1628,7 @@ bsdcpio_test_LDADD=libarchive_fe.la
 
 cpio/test/list.h: $(bsdcpio_test_SOURCES)
 	@$(MKDIR_P) cpio/test
-	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $^ | LC_COLLATE=C sort > $@
+	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $(bsdcpio_test_SOURCES) | LC_COLLATE=C sort > $@
 
 if BUILD_BSDCPIO
 bsdcpio_test_programs= bsdcpio_test
@@ -1751,7 +1751,7 @@ bsdcat_test_LDADD=libarchive_fe.la
 
 cat/test/list.h: $(bsdcat_test_SOURCES)
 	@$(MKDIR_P) cat/test
-	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $^ | LC_COLLATE=C sort > $@
+	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $(bsdcat_test_SOURCES) | LC_COLLATE=C sort > $@
 
 if BUILD_BSDCAT
 bsdcat_test_programs= bsdcat_test
@@ -1884,7 +1884,7 @@ bsdunzip_test_LDADD=libarchive_fe.la
 
 unzip/test/list.h: $(bsdunzip_test_SOURCES)
 	@$(MKDIR_P) unzip/test
-	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $^ | LC_COLLATE=C sort > $@
+	$(AM_V_GEN)grep -h '^DEFINE_TEST(' $(bsdunzip_test_SOURCES) | LC_COLLATE=C sort > $@
 
 if BUILD_BSDUNZIP
 bsdunzip_test_programs= bsdunzip_test
